python/lsst/pipe/base/butlerQuantumContext.py

```diff
@@ -97,6 +102,28 @@ def _put(self, value: Any, ref: DatasetRef):
         self._checkMembership(ref, self.allOutputs)
         self.__butler.put(value, ref)
 
+    def _issue_interval_log(self, next_log_time: float, msg: str, *args: Any) -> float:
+        """Issue a verbose log message if the interval has been exceeded.
+
+        Parameters
+        ----------
+        next_log_time : `float`
+            Time to issue the next message (as returned by `time.time()`).
+        msg : `str`
+            Message to issue if the time has been exceeded.
+        *args : Any
+            Parameters to be passed to the log system.
+
+        Returns
+        -------
+        next_log_time : `float`
+            Time to issue next log message.
+        """
+        if (current_time := time.time()) > next_log_time:
+            _LOG.verbose(msg, *args)
```

Comment:
_If_ we are moving it to the logging module, I think it'd also make more sense to make it `_LOG.log`, with level=logging.VERBOSE being passed in as a keyword argument

Reply:
While we are generalizing, I wonder if we can make it a coroutine so that the `next_log_time` value is preserved without having to pass it during calling this. Only the relevant `msg` has to be passed in; `next_log_time` will be preserved and `_LOG_INTERVAL` is just a constant defined at the module-level.